proj/src/main/webapp/script.js

```diff
@@ -366,4 +373,54 @@ function changeChatOnClick(domElement, tag, chatId) {
         dbRefObject = getDbRef(tag, chatId);
         initRef();
     });
+}
+
+function populateProfileSidebar() {
+    firebase.auth().onAuthStateChanged(function(user) {
+        if (user) {
+        const userRef = firebase.database().ref('/users/'+user.uid)
+        userRef.once('value', snap => {
+            const userObj = {};
+            userObj.photo = user.photoUrl;
+            userObj.uid = user.uid;
+            userObj.fname = snap.val().firstName;
+            userObj.lname = snap.val().lastName;
+            userObj.bio = snap.val().bio;  // there is no bio yet
+            userObj.tags = snap.val().allTags;
+            addUserInfoToDom(userObj)
+            })
+        }
+    });
+}
+
+function addUserInfoToDom(userObj) {
+    const profile = document.getElementById('user-profile');
+    profile.querySelector("#user-display-name").innerText = userObj.fname + ' ' + userObj.lname;
+    profile.querySelector("#user-pfp").src = userObj.photo;
+    profile.querySelector("#user-bio").innerText = userObj.bio;
+
+    const tagList = profile.querySelector("#user-tags");
+    for (tag in userObj.tags) {
+        if (userObj.tags.hasOwnProperty(tag)) {
+            const tagNode = document.createElement('li');
+            tagNode.innerText = tag;
+            tagList.appendChild(tagNode);
+        }
+    }
+}
+
+function switchToSettings() {
+    const settings = document.getElementById('user-profile');
+    const sidebar = document.getElementById('sidebar');
+
+    sidebar.classList.add("display-none");
+    settings.classList.remove("display-none");
+}
+
+function closeSettings() {
+    const settings = document.getElementById('user-profile');
+    const sidebar = document.getElementById('sidebar');
+
+    settings.classList.add("display-none");
+    sidebar.classList.remove("display-none");
```

Comment:
It looks like you're closing the settings and opening the sidebar? Should this sidebar always be open? Is the sidebar a list of all your chats and should it always be re-opened? If it is collapsible and it was collapsed before you clicked on settings, I would think it should stay collapsed.

Reply:
from the mockups that Lee showed us I think we want the sidebar to always be open, but I can double check